src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -215,6 +226,15 @@ public async Task InitializeUIAffinitizedServicesAsync(IAsyncServiceProvider asy
             await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(_threadingContext.DisposalToken);
 
             openFileTracker.ProcessQueuedWorkOnUIThread();
+
+            // Switch to a background thread to avoid loading option providers on UI thread.
+            await TaskScheduler.Default;
+
+            var telemetryService = (VisualStudioWorkspaceTelemetryService)Services.GetRequiredService<IWorkspaceTelemetryService>();
+            telemetryService.InitializeTelemetrySession(telemetrySession);
```

Comment:
could InitializeTelemetrySession itself just become async, and it can then switch to the BG do to this work?  @sharwell i believe you said that's the preferred way to do BG work?

Reply:
The problem isn't in `InitializeTelemetrySession` though. The problem is in accessing default TelemetrySession, which needs to happen on UI thread. Switching to background thread after that is to avoid doing the rest of the work on UI thread as well (that may include assembly loading due to access to options).

Reply:
i totally get that :)  i'm simply saying that in terms of coding pattern though, instead of having the switch here, just consider pushign it into InitializeTelemetrySession (which would need to become async).  Though i want to know @sharwell 's thought here

Reply:
I don't see the benefit nor what pattern do you mean.

Reply:
Make InitializeTelemetrySession async itself, and then move the switch to the BG into it.  That way it is the async method itself that decides this sort of thing instead of the caller.  In other words a caller doesn't need to know if it should push work on another component into the BG.  Instead it just makes hte await call on that component and that compnent itself can make that determination within it.

Reply:
I prefer the caller deciding this though since it had to decide to get the default telemetry session instance on UI thread. Why would the service care about switching to background thread? No other service we have that's using options does that.